### ISSUE TITLE
Feature Request:  Add support for TCS3725 ahd VEML6040 RGB Color Sensors

### ISSUE THREAD

**Issue by Jibun-no-Kage** [raspberrypi/linux#6446]
Feature Request:  Add support for TCS34725 ahd VEML6040 RGB Color Sensors as device tree overlays.

Example of TCS34725 RGB Color Sensor...
https://www.adafruit.com/product/1334

VEML6040 RGB Color Sensor...
https://www.taydaelectronics.com/veml6040-color-sensor-rgbw-i2c-module.html

**Comment by pelwell** [comment]
TCS34725 will need it's driver enabling - `CONFIG_TCS3472=m`.  It makes no use of Device Tree configuration, so it should just be a matter of adding the name to the `i2c-sensor` overlay.

The VEML6040 (not VMEML6040) driver doesn't exist in 6.6, but it is there in 6.12. It also doesn't seem to require any configuration.

If I have a moment I might throw together the necessary config and overlay updates against 6.12. What's your application?

**Comment by pelwell** [comment]
And which I2C addresses can the devices appear on - don't make me read the datasheets.

**Comment by Jibun-no-Kage** [comment]
Oh, right, TCS34725 is 0x29...  VEML6040 is 0x10...

The use case is that I am using these sensors to read a 'tally' light on a remote floating head camera, at the end of 25 or so foot  jib or boom crane... which is beyond visibility to anyone at ground level.  A micro controlller, i.e. in this case Pi Zero 2W forwards this color state to a production broadcast monitor at ground level, that has a custom GPIO interface,  I have not decided if I will use solid-state relays, opto-couplers, or NPN transistors, or even mosfet to do the actual GPIO pin state changes between the Pi and monitor GPIO interface yet.  but since i have the solid state relays, likely use those for the initial testing.,  And of course, the relays provide isolation between Pi GPIO and Production Monitor GPIO.

Sensor->i2c->Pi->Evaluate RGB levels->Pi GPIO->Trigger applicable relay.  On production broadcast monitor GPIO interface, Pin to GND, i.e. Low 'On', Pin High 'Off', for the respective color of tally status on-screen of monitor.  So I just via relay, short pin to GND for the given pin, i.e. color, red or green.

Typical broadcast switchers and cameras... Red is on-air, Green is stand-by or preview.  Or none... if camera is not pending preview or on-air.

TCS34725 is the first sensor I am testing, likely using initially, so VEML6040 in 6.12. is cool.  As always, thanks for doing this, greatly appreciated!

**Comment by Jibun-no-Kage** [comment]
You know... I could have sworn I posted this request to the 'firmware' section... Sorry about that.  I guess with all the page jumping I was doing... I did not  realize I was in the wrong section when I made the request.  Thanks for transferring it to the right section/queue.

**Comment by pelwell** [comment]
See https://github.com/raspberrypi/linux/pull/6448. Wait about 40 minutes for the autobuilds to complete, then run `sudo rpi-update pulls/6448` to install a trial kernel, being careful to back up any important data first.

**Comment by Jibun-no-Kage** [comment]
Yup... understood and well do.  Just a side question... ball park when this will go main stream?  Just curious.

**Comment by pelwell** [comment]
I missed a few back-port commits - it should be good now, so the build will be available to try by the morning.

Stable kernels (and hence apt packages) are usually updated roughly every three months. There is likely to be at least one more 6.6 release before we move to the next LTS kernel - probably 6.12.

**Comment by pelwell** [comment]
Let me know when you've tested #6448 and I'll merge it (if it works...)

**Comment by Jibun-no-Kage** [comment]
Did you finally merge it?  I completely forgot the update the comments here.

**Comment by pelwell** [comment]
I was waiting for feedback that never came. It's rebased now, so if you wait another ~15 minutes you'll be able to `sudo rpi-update pulls/6448` to test.

**Comment by Jibun-no-Kage** [comment]
Cool.... It is late now, but late tomorrow I should have some feed back to post.  Thanks again.
